### ISSUE TITLE
[TorchAgent] Allow agent to force the end of training.

### ISSUE THREAD

**Issue by stephenroller** [facebookresearch/ParlAI#2107]
We should define an exception, say `EndTrainingException` that can be raised by a TorchAgent in `train_step`. If so, the train_script should catch it and end training. This would be useful for learning rate schedulers that have an explicit end point, like Cosine or LinearDecay.

**Comment by stephenroller** [comment]
StopTraining might be a nicer word for it.

**Comment by stephenroller** [comment]
Fixed in #2138.
